### PR TITLE
Initial setup for new observIQ exporter

### PR DESCRIPTION
**Pull request by BinaryFissionGames**
**Description:**
This PR begins the implementation of an exporter for [observIQ](https://observiq.com/). It includes basic setup of the component with no implementation yet.

**Link to tracking Issue:** 
N/A

**Testing:**
Unit tests for the setup are implemented. 

**Documentation:**
exporter/observiqexporter/README.md documents input parameters and usage (sending logs to observiq)

### CODE REVIEW

File: exporter/observiqexporter/example/config.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+receiver:
+  filelog:
```
Comment (djaglowski) [inline exporter/observiqexporter/example/config.yaml:2]: Typically, we want these example configs to use `nop` for all other components.

[For example](https://github.com/open-telemetry/opentelemetry-collector/blob/3f5c7180c51ed67a6f54158ede5e523822e9659e/exporter/otlpexporter/testdata/config.yaml#L5-L9).

**Review by djaglowski — COMMENTED** [review]
Overall looks very nice. A few minor items to address though.

File: exporter/observiqexporter/example/config.yaml (hunk 1 above)
Comment (BinaryFissionGames) [reply exporter/observiqexporter/example/config.yaml:2]: This one was intended to be an example file for how to use it, not necessarily test data. For instance, both the [splunk](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/9c6bc6e00097d81ac59528d4b03fae88f26bb4f6/exporter/splunkhecexporter/example/otel-collector-config.yml) and [logzio](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/9c6bc6e00097d81ac59528d4b03fae88f26bb4f6/exporter/logzioexporter/example/config.yaml) exporters have these example configurations, so my intention was to mirror that (although a lot of the exporters don't seem to include an example config, it seems).

I was planning on putting in that testdata config with the implementation, in a later PR. Do you think that that config is necessary for this first commit?

File: exporter/observiqexporter/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+This exporter supports sending log data to [observIQ](https://observiq.com/)
+
+**This exporter is currently being developed and is not yet ready for use**
+
+## Configuration
+
+The following configuration options are required:
+
+- `api_key` (no default): The API key used to authenticate with observIQ. See the (docs for the settings page)[https://docs.observiq.com/docs/overview#settings-page] to generate one.
```
[the remark this answers is not in the document]
Comment (BinaryFissionGames) [reply exporter/observiqexporter/README.md]: Should be fixed!

File: exporter/observiqexporter/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+The following configuration options can also be configured:
+
+- `endpoint` (default "https://nozzle.app.observiq.com/v1/add"): Endpoint where logs are sent to over http(s).
```
[the remark this answers is not in the document]
Comment (BinaryFissionGames) [reply exporter/observiqexporter/README.md]: Put backticks around all default options.

File: exporter/observiqexporter/example/config.yaml (hunk 1 above)
Comment (djaglowski) [reply exporter/observiqexporter/example/config.yaml:2]: Ah, my mistake. I didn't look closely enough and presumed that this was in the `testdata` directory. Looks good.

**Review by djaglowski — APPROVED** [review]

**Pull request merged**